python/cudf/cudf/core/subword_tokenizer.py

```diff
@@ -0,0 +1,279 @@
+        attention_mask: (type defined by return_tensors)
+            A tensor of indices specifying which tokens
+            should be attended to by the model
+        metadata: (type defined by return_tensors)
+            Each row contains the index id of the original string and the
+            first and last index of the token-ids that are non-padded and
+            non-overlapping
+
+        Examples
+        --------
+        >>> import cudf
+        >>> from cudf.utils.hash_vocab_utils  import hash_vocab
+        >>> hash_vocab('bert-base-cased-vocab.txt', 'voc_hash.txt')
+
+        >>> from cudf.core.subword_tokenizer import SubwordTokenizer
+        >>> cudf_tokenizer  = SubwordTokenizer('voc_hash.txt',
+                                                do_lower_case=True)
+
+        >>> str_series = cudf.Series(['This is the', 'best book'])
+        >>> tokenizer_output = cudf_tokenizer(str_series,
+                            max_length=8,
+                            max_num_rows=len(str_series),
+                            padding='max_length',
+                            return_tensors='pt',
+                            truncation=True)
+        >>>tokenizer_output['input_ids']
+        tensor([[ 101, 1142, 1110, 1103,  102,    0,    0,    0],
+                [ 101, 1436, 1520,  102,    0,    0,    0,    0]],
+                device='cuda:0',
+               dtype=torch.int32)
+        >>>tokenizer_output['attention_mask']
+        tensor([[1, 1, 1, 1, 1, 0, 0, 0],
+                [1, 1, 1, 1, 0, 0, 0, 0]],
+                device='cuda:0', dtype=torch.int32)
+
+        >>>tokenizer_output['metadata']
+        tensor([[0, 1, 3],
+                  [1, 1, 2]], device='cuda:0', dtype=torch.int32)
+       """
+
+        if return_token_type_ids:
+            # raise not currently supported
+            # Can also return zeros
+            error_msg = "Returning token_type_ids is currently supported"
+            raise NotImplementedError(error_msg)
+
+        if truncation in (False, "do_not_truncate"):
+            truncation = False
+            warning_msg = (
+                "When truncation is not True, the behaviour currently differs "
+                "from HuggingFace as cudf always returns overflowing tokens"
+            )
+            warn(warning_msg)
+            if add_special_tokens:
+                error_msg = (
+                    f"Adding special tokens is not supported"
+                    f"with truncation = {truncation}. "
+                )
+                recommendation = (
+                    "Custom Cupy kernel can potentially"
+                    "be used to add it. For reference "
+                    "see: _bert_add_special_tokens "
+                )
+                raise NotImplementedError(error_msg + recommendation)
+
+        if padding != "max_length":
+            error_msg = (
+                "Currently padding to the provided max_length is supported"
```

Comment:
Should this be *not* supported?
```suggestion
                "Currently padding to the provided max_length is not supported"
```

Reply:
Nope, it the only value with support.  I changed the error message to below with commit https://github.com/rapidsai/cudf/commit/0018779bde5984931e114a3cd412ce869706a7b3 now to make it clearer. : 

```
Only padding to the provided max_length is currently supported
```